src/NSubstitute/Arg.cs

```diff
@@ -14,7 +14,14 @@ namespace NSubstitute
     /// </summary>
     public static class Arg
     {
-        public class AnyType
+        /// <summary>
+        /// This type can be used with any matcher to match a generic type parameter.
+        /// </summary>
+        /// <remarks>
+        /// If the generic type parameter has constraints, you will have to create a derived class/struct that
+        /// implements those constraints.
+        /// </remarks>
+        public interface AnyType
```

Comment:
**thought:** what do you think about providing implementations for [common constraints](https://learn.microsoft.com/en-us/dotnet/csharp/programming-guide/generics/constraints-on-type-parameters)? In particular `AnyClass` (with default ctor to match `: class, new()`), and `AnyStruct` or `AnyValueType`.